### DISCUSSION TITLE
Add Gato GraphQL as source?

### DISCUSSION THREAD

**Discussion by leoloso** [colbyfayock/next-wordpress-starter#574] [General]
Hi @colbyfayock 

I've released Gato GraphQL in the WordPress plugin directory: https://wordpress.org/plugins/gatographql/

This is yet another GraphQL server for WordPress, like WPGraphQL.

I am using using Gato GraphQL as the source for your starter project, and it works wonders:

https://github.com/leoloso/next-wordpress-starter/tree/Add-plugin-GraphQL-API-for-WP-as-the-GraphQL-source-2

Question: Would you be interested in incorporating it into the starter (in addition to WPGraphQL), allowing developers to choose which source to use on their projects?

---

The changes to adapt the code have been minimal: Only the GraphQL queries in the data files had to be adapted, and no JS logic whatsoever. You can see the differences here:

https://github.com/colbyfayock/next-wordpress-starter/compare/main...leoloso:Add-plugin-GraphQL-API-for-WP-as-the-GraphQL-source-2#files_bucket

The way I see it, all the data files could be located under a subfolder for the provider (`wpgraphql` and `gatographql`), using WPGraphQL as the default one.

The supported WordPress GraphQL schema is the same (with a few different field names). Yoast SEO is not integrated though.

**Comment by colbyfayock** [comment]
hey @leoloso, im not opposed to the idea but im not sure realistically what kind of timeline we'd be looking at for this support

i've been a bit MIA on this project admittedly but i've already started on a major overhaul of the project that i have yet to jump back in and pick up: https://github.com/colbyfayock/next-wordpress-starter/pull/477

so if anything it owuld likely be after the App Router migration and not sure when that will happen

**Comment by leoloso** [comment]
@colbyfayock I guess there's no rush to incorporate it. However I can help, then we can make it sooner.

I don't mind doing the changes in my repo, then whenever they are ready we can merge it to your starter, either before or after the App Router. In that case, I need your direction and supervision, and I implement it.

I'd need to know if you agree with the proposal of having a folder for providers, with subfolders `wpgraphql` and `gatographql`, and move there all the data files containing the queries.

What do you think?

**Comment by leoloso** [comment]
@colbyfayock any feedback on my previous comment? I have a bit of time now and I could work on this

**Reply by colbyfayock** [reply]
hey @leoloso it seems reasonable but im not ready to commit to anything until i have time to first revisit this project and complete the App Router work, or merge a state of it into the main branch. it's been out of my context for a while so i need to have time to spin back up on it and understand the potential implications

**Reply by leoloso** [reply]
Alright @colbyfayock, I went ahead and implemented it all by myself 😃 

Full explanations here: https://github.com/leoloso/next-wordpress-starter/pull/1

I'll be pointing to this starter project on the [Gato GraphQL website](https://gatographql.com), so my users can use the starter.

If eventually you'd like to merge these changes into your project, that would be amazing (and I will certainly promote it). Of course I can help, just let me know.

Cheers

**Reply by colbyfayock** [reply]
nice! thanks for sharing. when i loop back around to working on this again ill check that out!
